### ISSUE TITLE
Adding RowNum to queryStream

### ISSUE THREAD

**Issue by mdunisch** [oracle/node-oracledb#633]
I currently testing using the `queryStream`-Method to handle large data from our Database.

Because you set every row on the *data-Event* of the stream this is extremely slow and unusable. I wonder why this isn't a parameter. All node-Stream-Functions allow you to definie sizes of buffer.

I would suggest to add a paramter like:

`
queryStream(String sql, [Object bindParams], [Object options], [Number RowsPerData]);
`

**Comment by cjbj** [comment]
What are the perf results you are seeing for `queryStream()` vs ResultSet `getRows()`?

**Comment by dmcghan** [comment]
@mdunisch Have you seen [the section on row prefetching](https://github.com/oracle/node-oracledb/blob/master/doc/api.md#-917-row-prefetching)? The row prefetch can be used to reduce the number of round trips between the database and Node.js by buffering more data on the Node.js server. 

In theory, this should have a much bigger impact on performance than reducing function invocations in Node.js (though that would be nice too). Try adjusting that setting and let us know how it goes.

**Comment by cjbj** [comment]
@dmcghan good point.  I'd also like to see some user `queryStream()` benchmarks for various values of [`maxRows`](https://github.com/oracle/node-oracledb/blob/master/doc/api.md#propdbmaxrows) and the prefetch set in the region of 0 - 2.

**Comment by cjbj** [comment]
For a small benchmark with v2 see https://blogs.oracle.com/opal/node-oracledb-v2-query-methods-and-fetch-tuning

**Comment by dmcghan** [comment]
@mdunisch I realize you've long moved on from this issue, but I just saw it again while going through the issues in the [poll](https://pt.surveymonkey.com/r/8LT889R). This likely has to do with excessive event loop iterations.

You mention that "All node-Stream-Functions allow you to definie sizes of buffer." Could you please point to an example that we could look at? An object stream would be best. 

I like the idea, but I want to make sure we do something based on standards.
